tests/MongoDB.Bson.Tests/Specifications/bson-corpus/BsonCorpusTestRunner.cs

```diff
@@ -0,0 +1,358 @@
+﻿/* Copyright 2020 MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using FluentAssertions;
+using MongoDB.Bson.IO;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Serializers;
+using MongoDB.Bson.TestHelpers.JsonDrivenTests;
+using Xunit;
+
+namespace MongoDB.Bson.Tests.Specifications.bson_corpus
+{
+    public class BsonCorpusTestRunner
+    {
+        [SkippableTheory]
+        [ClassData(typeof(TestCaseFactory))]
+        public void RunTestDefinition(JsonDrivenTestCase testCase)
+        {
+            var test = testCase.Test;
+            var testType = test["type"].AsString;
+            var shared = testCase.Shared;
+
+            JsonDrivenHelper.EnsureAllFieldsAreValid(
+                shared,
+                "_path",
+                "description",
+                "bson_type",
+                "test_key",
+                "deprecated",
+                "valid",
+                "decodeErrors",
+                "parseErrors");
+
+            if (shared.Contains("bson_type"))
+            {
+                // This field is gnored
+            }
+
+            if (shared.Contains("test_key"))
+            {
+                // This field is gnored
+            }
+
+            if (shared.Contains("deprecated"))
+            {
+                // This field is gnored
+            }
+
+            switch (testType)
```

Comment:
With the helper variable assigned closer to where it is needed this would be:

    var testType = test["type"].AsString;
    switch (testType)

Reply:
Done.